okhttp-android/src/main/kotlin/okhttp3/android/AndroidAsyncDns.kt

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2022 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package okhttp3.android
+
+import android.net.DnsResolver
+import android.net.Network
+import android.os.Build
+import androidx.annotation.RequiresApi
+import java.net.InetAddress
+import java.net.UnknownHostException
+import java.util.concurrent.Executors
+import okhttp3.AsyncDns
+
+/**
+ * Dns implementation based on android.net.DnsResolver, which submits two requests for
+ * A and AAAA records, and returns the addresses or exception from each before returning.
+ *
+ * @param network network to use, if not selects the default network.
+ */
+@RequiresApi(Build.VERSION_CODES.Q)
+class AndroidAsyncDns(
```

Comment:
Hide public constructor?

Reply:
(Hmmm . . . . actually for `Network` this is useful . . .  hmmmm)

Reply:
Yep - I actually liked this bit. There are bugs in stackoverflow because android/wear networking returns differing/empty results across Wifi, LTE, Bluetooth paired connections (on Wear).

The choice is implicit currently with Dns.SYSTEM, and this makes it explicit.